docs/from.md

```diff
@@ -0,0 +1,161 @@
+  ...iter.map(dependencies.entries(), ([name, ver]) => searchDeepDependencies(`${name}@${ver}`, null, configRef))
+];
+
+for await (const dep of combineAsyncIterators({}, ...iterators)) {
+  yield dep;
+}
+```
+
+### 2.2
+
+Everything is done using Async Generators which make everything more simple by flattening dependency. This step use the same Manifest API from pacote. If a given package still have dependencies, then the recursive function will continue to execute until there is no more (or it will stop if the maximum depth has been reach).
+
+Here is a simplified version of the Generator function:
+
+```js
+export async function* searchDeepDependencies(packageName, gitURL, options) {
+  const { exclude, currDepth = 0, parent, maxDepth } = options;
+
+  const { name, version } = await pacote.manifest(gitURL ?? packageName, {
+    ...NPM_TOKEN,
+    registry: getLocalRegistryURL(),
+    cache: `${os.homedir()}/.npm`
+  });
+  const { dependencies, customResolvers } = mergeDependencies(pkg);
+
+  const current = new Dependency(name, version, parent);
+  if (currDepth !== maxDepth) {
+    const config = {
+      exclude, currDepth: currDepth + 1, parent: current, maxDepth
+    };
+
+    const depsNames = await Promise.all(iter.map(dependencies.entries(), getCleanDependencyName));
+    for (const [fullName, cleanName, isLatest] of depsNames) {
+      yield* searchDeepDependencies(fullName, null, config);
+    }
+  }
+
+  yield current;
+}
+```
+
+Each time a dependency is retrieved the code will run two separate analyses in parallel:
+- Tarball scanning (using pacote.extract)
+- Fetching additional metadata on the registry
+
+## Steps 3: Apply vulnera strategy
+
+The third step is to recover the list of vulnerabilities using the active [vulnera](https://github.com/NodeSecure/vulnera) strategy (or nothing if no strategy is chosen).
+
+```js
+const { hydratePayloadDependencies, strategy } = await vuln.setStrategy(vulnerabilityStrategy);
+await hydratePayloadDependencies(dependencies, {
+  useStandardFormat: true,
+  path: location
+});
+```
+
+For more information on how to create a strategy and how they operate, please [read the following documentation](https://github.com/NodeSecure/vulnera/blob/main/docs/adding_new_strategy.md).
+
+## Steps 4: Get warnings and flagged authors
+
+In this step we look for packages or authors potentially identified as problematic (by us or the person requesting the analysis).
+
+```js
+const { warnings, flaggedAuthors } = await getDependenciesWarnings(dependencies);
+payload.warnings = warnings;
+payload.flaggedAuthors = flaggedAuthors;
+```
+
+> Under the hood we use our newest package [@nodesecure/authors](https://github.com/NodeSecure/authors) to optimize and flag authors.
+
+By default we show warnings for the following two packages:
+- @scarf/scarf
+- iohook
+
+Since the incident with Faker we also identify Marak's packages as dangerous.
```

Comment:
May we could add a link to this incident (there is also the DoS in colors package) ?
Like: https://snyk.io/blog/npm-faker-package-open-source-libraries/ or https://blog.sonatype.com/faker-npm-library-gets-new-home